src/pulsar_producer.erl

```diff
@@ -177,7 +177,8 @@ init({PartitionTopic, Server, ProxyToBrokerUrl, ProducerOpts0}) ->
             false ->
                 #{mem_only => true};
             BaseDir ->
-                Dir = filename:join([BaseDir, PartitionTopic]),
+                PartitionTopic64 = base64:encode_to_string(PartitionTopic),
+                Dir = filename:join([BaseDir, PartitionTopic64]),
```

Comment:
Sorry, this will lose connection between the topic and the data directory (in case one needs to do manual maintenance). 
What I meant was just to "escape" special chars like `/` in the name, one way is to base64 encode only the special chars. (however may lead to clash).

Reply:
maybe url-encode is a better option.

Reply:
```
2> edoc_lib:escape_uri("/root").
"%2froot"
3>
```
the old http_uri api was deprecated

Reply:
unfortunately `%` is not allowed in windows.

Reply:
seems to be safe if we use `=` instead of `%` for escaping.
```
iolist_to_binary(string:replace(edoc_lib:escape_uri("/root"), "%", "=")).
<<"=2froot">>
```

Reply:
:heavy_check_mark: 

I needed to copy that function from `edoc_lib` because dialyzer was refusing to see that private function :see_no_evil: